extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryChangeLogSource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.streams.querymanager;
+
+import java.util.Set;
+
+import org.apache.rya.streams.api.queries.QueryChangeLog;
+
+import com.google.common.util.concurrent.Service;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A source for QueryChangeLogs.
+ * Subscribers to this source will be notified when QueryChangeLogs are created/deleted
+ * <p>
+ * When {@link QueryChangeLog}s are found, the {@link SourceListener}s are
+ * notified with those logs.
+ */
+@DefaultAnnotation(NonNull.class)
+public interface QueryChangeLogSource extends Service {
+
+    /**
+     * Adds a {@link SourceListener} to be notified when {@link QueryChangeLog}s
+     * have been found.
+     * <p>
+     * When a new subscriber is added, it will be notified with the existing {@link QueryChangeLog}s.
+     *
+     * @param listener - The {@link SourceListener} to add. (not null)
+     */
+    public void subscribe(final SourceListener listener);
+
+    /**
+     * Removes a {@link SourceListener}.
+     *
+     * @param listener - The {@link SourceListener} to remove. (not null)
+     */
+    public void unsubscribe(final SourceListener listener);
+
+    /**
+     * A listener that is notified when a {@link QueryChangeLog} has
+     * been added or removed from a {@link QueryChangeLogSource}.
+     */
+    public interface SourceListener {
+        /**
+         * Notifies the subscriber that there are new {@link QueryChangeLog}s in the
+         * {@link QueryChangeLogSource}.
+         *
+         * @param logs - The new {@link QueryChangeLog}s. (not null)
+         */
+        public void notifyCreate(final Set<QueryChangeLog> logs);
```

Comment:
In hindsight, I don't know that this is enough information for the QueryManager to do its work. Without knowing which Rya Instance the change log represents, the query manager will not know what source topics to use when building a topology to run the query.